internal/app/machined/pkg/controllers/network/link_config.go

```diff
@@ -162,6 +162,12 @@ func (ctrl *LinkConfigController) Run(ctx context.Context, r controller.Runtime,
 		if cfgProvider != nil {
 			for _, device := range cfgProvider.Machine().Network().Devices() {
 				configuredLinks[device.Interface()] = struct{}{}
+
+				if device.Bond() != nil {
```

Comment:
bond slaves are implicitly configured even if they're not mentioned in the configuration explicitly